examples/asr/jasper_an4_debug.py

```diff
@@ -0,0 +1,300 @@
+    total_steps = steps_per_epoch * args.num_epochs
+    logging.info("Train samples=", num_samples, "num_steps=", total_steps)
+
+    # # Create a data_layer for evaluation.
+    # data_layer_eval = nemo_asr.AudioToTextDataLayer.import_from_config(
+    #     model_config_file, "AudioToTextDataLayer_eval", overwrite_params={"manifest_filepath": args.eval_datasets},
+    # )
+
+    # num_samples = len(data_layer_eval)
+    # logging.info(f"Eval samples={num_samples}")
+
+    # Instantiate data processor.
+    data_preprocessor = nemo_asr.AudioToMelSpectrogramPreprocessor.import_from_config(
+        model_config_file, "AudioToMelSpectrogramPreprocessor"
+    )
+
+    # Instantiate JASPER encoder-decoder modules.
+    jasper_encoder = nemo_asr.JasperEncoder.import_from_config(model_config_file, "JasperEncoder")
+    jasper_decoder = nemo_asr.JasperDecoderForCTC.import_from_config(
+        model_config_file, "JasperDecoderForCTC", overwrite_params={"num_classes": len(vocab)}
+    )
+
+    # Instantiate losses.
+    ctc_loss = nemo_asr.CTCLossNM(num_classes=len(vocab))
+    greedy_decoder = nemo_asr.GreedyCTCDecoder()
+
+    # Create a training graph.
+    audio, audio_len, transcript, transcript_len = data_layer()
+    processed, processed_len = data_preprocessor(input_signal=audio, length=audio_len)
+    encoded, encoded_len = jasper_encoder(audio_signal=processed, length=processed_len)
+    log_probs = jasper_decoder(encoder_output=encoded)
+    predictions = greedy_decoder(log_probs=log_probs)
+    loss = ctc_loss(log_probs=log_probs, targets=transcript, input_length=encoded_len, target_length=transcript_len)
+
+    # # Create an evaluation graph.
+    # audio_e, audio_len_e, transcript_e, transcript_len_e = data_layer_eval()
+    # processed_e, processed_len_e = data_preprocessor(input_signal=audio_e, length=audio_len_e)
+    # encoded_e, encoded_len_e = jasper_encoder(audio_signal=processed_e, length=processed_len_e)
+    # log_probs_e = jasper_decoder(encoder_output=encoded_e)
+    # predictions_e = greedy_decoder(log_probs=log_probs_e)
+    # loss_e = ctc_loss(
+    #     log_probs=log_probs_e, targets=transcript_e, input_length=encoded_len_e, target_length=transcript_len_e,
+    # )
+    logging.info("Num of params in encoder: {0}".format(jasper_encoder.num_weights))
+
+    # Callbacks to print info to console and Tensorboard.
+    # train_callback = nemo.core.SimpleLossLoggerCallback(
+    #     tensors=[loss, predictions, transcript, transcript_len],
+    #     print_func=partial(monitor_asr_train_progress, labels=vocab),
+    #     get_tb_values=lambda x: [["loss", x[0]]],
+    #     tb_writer=nf.tb_writer,
+    # )
+
+    # loss.rename("test")
+    # train_callback = nemo.core.SimpleLossLogger(tensors_to_log=["test"])
+
+    # train_callback = nemo.core.SimpleLossLogger()
+    predictions.rename("test")
+    train_callback = nemo.core.SimpleLossLogger(tensors_to_log=["loss", "test"])
```

Comment:
just wondering - what is the advantage of passing "names" instead of NmTensors?

Reply:
I have no good answer, except that I think strings are easier for users to work with inside of their own callback functions.
On the other hand, NmTensors would make creating this PR easier.